### PR TITLE
Regex segment local evaluation with int trait

### PR DESCRIPTION
**Pull request by floranpagliai**
#71

### CODE REVIEW

File: src/Engine/Segments/SegmentConditionModel.php [hunk 1]
```diff
@@ -101,7 +101,7 @@ public function matchesTraitValue($traitValue): bool
 
         switch ($this->operator) {
             case (SegmentConditions::EQUAL):
-                $condition = $traitValue == $castedValue;
+                $condition = $traitValue === $castedValue;
```
Comment (floranpagliai) [inline src/Engine/Segments/SegmentConditionModel.php:104]: As value are cast on line 91, might as well make a strict comparison

File: src/Engine/Segments/SegmentConditionModel.php [hunk 2]
```diff
@@ -116,7 +116,7 @@ public function matchesTraitValue($traitValue): bool
                 $condition = $traitValue <= $castedValue;
                 break;
             case (SegmentConditions::NOT_EQUAL):
-                $condition = $traitValue != $castedValue;
+                $condition = $traitValue !== $castedValue;
```
Comment (floranpagliai) [inline src/Engine/Segments/SegmentConditionModel.php:119]: Same here

File: src/Engine/Segments/SegmentConditionModel.php [hunk 3]
```diff
@@ -130,7 +130,7 @@ public function matchesTraitValue($traitValue): bool
                 break;
             case (SegmentConditions::IN):
                 if (in_array($traitValueType, ['string', 'integer'])) {
-                    $condition = in_array((string) $traitValue, explode(',', (string) $this->value));
+                    $condition = in_array((string) $traitValue, explode(',', (string) $this->value), true);
```
Comment (floranpagliai) [inline src/Engine/Segments/SegmentConditionModel.php:133]: Same

File: tests/Engine/Unit/Segments/SegmentModelsTest.php [hunk 4]
```diff
@@ -74,6 +74,7 @@ public function conditionParametersTraitValues()
             [SegmentConditions::REGEX, 'foo', '[a-z]+', true],
             [SegmentConditions::REGEX, 'FOO', '[a-z]+', false],
             [SegmentConditions::REGEX, '1.2.3', '\\d', true],
+            [SegmentConditions::REGEX, 123, '^\\d*[13579]$', true],
```
Comment (floranpagliai) [inline tests/Engine/Unit/Segments/SegmentModelsTest.php:77]: I added a test that check that an Integer trait value can be compared against a regex

**Review by matthewelwell — APPROVED** [review]
Thanks @floranpagliai !

**Comment by matthewelwell** [comment]
@floranpagliai I have just released this fix in v4.3.1 .

**Comment by floranpagliai** [comment]
Thank you so much @matthewelwell for the quick review and release!

**Pull request merged**